### PR TITLE
No Mod Section if You Are the Only Mod

### LINKED ISSUE #708: In the Purchase modal, move moderator error message to the Moderator box
Move the `You must select a moderator to make a Moderated Payment.` error message to the Moderator box, since that is where the list of moderator they need to select from are (or will be if they're loading).

image

### PR DESCRIPTION
**Pull request by jjeffryes**
In the purchase screen, the check to not show the moderator section was not accounting for situations where you were the only moderator, and would show a blank list of moderators.

This changes the view and template to use the same list, after the view has removed any IDs that shouldn't be in the list by passing hasModerators in as a template parameter instead of calculating it separately in the template.

This also moves the moderator error to the moderator section, and closes #708.

### CODE REVIEW

File: js/templates/modals/purchase/purchase.html [hunk 1]
```diff
@@ -97,9 +95,10 @@ <h2 class="h4 required"><%= ob.polyT('purchase.paymentTypeTitle') %></h2>
           <% } %>
         </div>
       </section>
-      <% if (hasModerators) { %>
+      <% if (ob.hasModerators) { %>
         <section class="contentBox padMd clrP clrBr clrSh3 showIfPay js-moderator">
           <h2 class="h4"><%= ob.polyT('purchase.moderatorTitle') %></h2>
+          <div class="js-moderated-errors"></div>
```
Comment (rmisio) [inline js/templates/modals/purchase/purchase.html]: Why was this added if nothing else was added to handle this? Is this the remnants of a refactor?

Comment (jjeffryes) [reply js/templates/modals/purchase/purchase.html]: I forgot, that was to fix #708 which is what I was originally working on when I noticed the mod section could show when there were no valid mods.
